Possible boss for the game.

DNQ4 opened the pull request:
Currently working on animationsprite for the boss and possble ways for the boss to attack the player. But if you guys think this is a good sprite for the boss or have another option let me know.


update 2024-03-6
Just made it so that the animationsprite would flip horizontal towards the player.
Also had to make two animationsprite because when it would flip horizontal, the sprite would be outside of its hitbox.

DNQ4 committed:
A possible boss for the game.

Currently working on animationsprite for the boss and possble ways for the boss to attack the player. But if you guys think this is a good sprite for the boss or have another option let me know.

Snurrigt1 commented:
This sprite will work

xix-xeaon commented on boss/Demon_boss.tscn:
```diff
@@ -0,0 +1,87 @@
+
+[ext_resource type="Texture2D" uid="uid://dlgqlt3riuxhm" path="res://boss/demon-Files/PNG/demon-idle.png" id="1_pjcdw"]
+
+[sub_resource type="GDScript" id="GDScript_iwou8"]
+script/source = "extends CharacterBody2D
+
+@export var speed = 300
+
+func _physics_process(delta):
+	var direction = global_position.direction_to(Global.player.global_position)
+	velocity = direction * speed 
+	move_and_slide()
+	
+	
+	
+		
+"
```
Looks like you're using an embedded script, it should use the .gd-file I think.

xix-xeaon reviewed (changes requested):
Do we need/want the GIF and PSD files?

DNQ4 committed:
Basic animations for the boss

Just made it so that the animationsprite would flip horizontal towards the player.

Also had to make two animationsprite because when it would flip horizontal, the sprite would be outside of its hitbox.

DNQ4 committed:
Merge branch 'xixs-students:main' into main

DNQ4 committed:
Delet useless files

DNQ4 committed:
Merge branch 'main' of https://github.com/DNQ4/IT23-game

Snurrigt1 commented:
Delete the empty folders (PSD, GIF), and change the folder name from 'boss' to 'Boss'.

You don't need two separate animations for left and right, you can use %AnimatedSprite2D.flip_h = true/false. Also, the animations has different FPS settings, personally I think 6 FPS looks better. 

Change motion mode from 'Grounded' to 'Floating'.

lawin666 commented:
I think motion mode should be floating because the boss is getting stuck with the player when they are near each other. 
Layers should be fixed too. You can look at mob.tscn and copy that one.

DNQ4 committed:
Just some small changes that you guys requested.

Changed the motion mode from Grounded to Floating and changed layers.

Snurrigt1 commented:
The boss needs to be slower
